vaadin-text-field-flow-parent/vaadin-text-field-flow/src/main/java/com/vaadin/flow/component/textfield/TextField.java

```diff
@@ -490,6 +498,16 @@ public Validator<String> getDefaultValidator() {
         return (value, context) -> getValidationSupport().checkValidity(value);
     }
 
+    @Override
+    public Registration addValidationStatusChangeListener(
+            ValidationStatusChangeListener<String> listener) {
+        return addClientValidatedEventListener(event -> {
+            listener.validationStatusChanged(
+                    new ValidationStatusChangeEvent<String>(this,
+                            !isInvalid()));
```

Comment:
The purpose of the second parameter of `ValidationStatusChangeEvent` is not very clear since it is not used anywhere.